Modules/posixmodule.c

```diff
@@ -6640,7 +6640,15 @@ static PyObject *
 posix_getgrouplist(PyObject *self, PyObject *args)
 {
 #ifdef NGROUPS_MAX
-#define MAX_GROUPS NGROUPS_MAX
+    /*
+     * NGROUPS_MAX is defined by POSIX.1 as the maximum
+     * number of supplimental groups a users can belong to.
+     * We have to increment it by one because
+     * getgrouplist() returns both the supplemental groups
+     * and the primary group, i.e. all of the groups the
+     * user belongs to.
+     */
+#define MAX_GROUPS (NGROUPS_MAX + 1)
```

Comment:
`MAX_GROUPS` is also defined later in `posixmodule.c` in `os_getgroups_impl`.   Without also changing it, the fix here can cause a compile warning:

```
../../source/Modules/posixmodule.c:6831:9: warning: 'MAX_GROUPS' macro redefined [-Wmacro-redefined]
#define MAX_GROUPS NGROUPS_MAX
        ^
../../source/Modules/posixmodule.c:6746:9: note: previous definition is here
#define MAX_GROUPS (NGROUPS_MAX + 1)
        ^
```
The simplest fix is to also change the latter definition to be `NGROUPS_MAX + 1.`

Reply:
I have made the requested changes; please review again